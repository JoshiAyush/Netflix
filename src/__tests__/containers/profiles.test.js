import React from "react";
import { render } from "@testing-library/react";

import { SelectProfileContainer } from "../../containers/index.js";

describe("<SelectProfile />", () => {
  it("renders the <SelectProfile /> with populated data", () => {
    const { container, getByText, getByTestId } = render(
      <SelectProfileContainer.Inner>
        <SelectProfileContainer.Title>
          Who's watching?
        </SelectProfileContainer.Title>

        <SelectProfileContainer.List>
          <SelectProfileContainer.User onClick={() => {}}>
            <SelectProfileContainer.Picture
              src="/images/users/1.png"
              alt="user"
              data-testid="profile-picture"
            />

            <SelectProfileContainer.Name>
              Ayush Joshi
            </SelectProfileContainer.Name>
          </SelectProfileContainer.User>
        </SelectProfileContainer.List>
      </SelectProfileContainer.Inner>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId("profile-picture")).toBeTruthy();
    expect(getByText("Ayush Joshi")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <SelectProfile /> with populated data but with misc profile image", () => {
    const { container, getByText, getByTestId } = render(
      <SelectProfileContainer.Inner>
        <SelectProfileContainer.Title>
          Who's watching?
        </SelectProfileContainer.Title>

        <SelectProfileContainer.List>
          <SelectProfileContainer.User onClick={() => {}}>
            <SelectProfileContainer.Picture data-testid="profile-picture-misc" />

            <SelectProfileContainer.Name>
              Ayush Joshi
            </SelectProfileContainer.Name>
          </SelectProfileContainer.User>
        </SelectProfileContainer.List>
      </SelectProfileContainer.Inner>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId("profile-picture-misc")).toBeTruthy();
    expect(getByText("Ayush Joshi")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
