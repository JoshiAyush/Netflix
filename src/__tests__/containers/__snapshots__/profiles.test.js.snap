// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectProfile /> renders the <SelectProfile /> with populated data 1`] = `
<div
  class="style__Inner-sc-1yu22yp-6 ewpSaC"
>
  <h1
    class="style__Title-sc-1yu22yp-1 fpbFLq"
  >
    Who's watching?
  </h1>
  <ul
    class="style__List-sc-1yu22yp-2 bMYplF"
  >
    <li
      class="style__Item-sc-1yu22yp-5 eKxBGc"
    >
      <img
        alt="user"
        class="style__Picture-sc-1yu22yp-4 jCYpVA"
        data-testid="profile-picture"
        src="/images/users/1.png"
      />
      <p
        class="style__Name-sc-1yu22yp-3 bKLGAC"
      >
        Ayush Joshi
      </p>
    </li>
  </ul>
</div>
`;

exports[`<SelectProfile /> renders the <SelectProfile /> with populated data but with misc profile image 1`] = `
<div
  class="style__Inner-sc-1yu22yp-6 ewpSaC"
>
  <h1
    class="style__Title-sc-1yu22yp-1 fpbFLq"
  >
    Who's watching?
  </h1>
  <ul
    class="style__List-sc-1yu22yp-2 bMYplF"
  >
    <li
      class="style__Item-sc-1yu22yp-5 eKxBGc"
    >
      <img
        class="style__Picture-sc-1yu22yp-4 jCYpVA"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="style__Name-sc-1yu22yp-3 bKLGAC"
      >
        Ayush Joshi
      </p>
    </li>
  </ul>
</div>
`;
